arbitrator/prover/src/lib.rs

```diff
@@ -328,17 +332,18 @@ pub unsafe extern "C" fn arbitrator_set_preimage_resolver(
     (*mach).set_preimage_resolver(Arc::new(
         move |context: u64, ty: PreimageType, hash: Bytes32| -> Option<CBytes> {
             if let PreimageType::EthVersionedHash = ty {
```

Comment:
You could slightly simplify this as `if ty == PreimageType::EthVersionedHash` but either way works